monkey/monkey_island/cc/resources/auth/registration.py

```diff
@@ -1,26 +1,42 @@
 import json
+import logging
 
 import flask_restful
 from flask import make_response, request
 
 import monkey_island.cc.environment.environment_singleton as env_singleton
 import monkey_island.cc.resources.auth.password_utils as password_utils
 from common.utils.exceptions import InvalidRegistrationCredentialsError, RegistrationNotNeededError
+from monkey_island.cc.database import mongo
 from monkey_island.cc.environment.user_creds import UserCreds
+from monkey_island.cc.setup.mongo.database_initializer import init_collections
+
+logger = logging.getLogger(__name__)
 
 
 class Registration(flask_restful.Resource):
     def get(self):
-        return {"needs_registration": env_singleton.env.needs_registration()}
+        is_registration_needed = env_singleton.env.needs_registration()
+        if is_registration_needed:
+            # if registration is required, drop previous user's data (for credentials reset case)
+            _drop_mongo_db()
```

Comment:
All this logic should be moved into `init_collections` method which gets called during registration. This way interaction with mongodb during registration is separated from the logic of resources. In general resources shouldn't interact with database directly.